include/arch/arm/aarch32/exc.h

```diff
@@ -48,14 +48,10 @@
 #endif
 
 #define _EXC_FAULT_PRIO 0
-#ifdef CONFIG_ZERO_LATENCY_IRQS
 #define _EXC_ZERO_LATENCY_IRQS_PRIO 0
-#define _EXC_SVC_PRIO 1
-#define _IRQ_PRIO_OFFSET (_EXCEPTION_RESERVED_PRIO + 1)
-#else
-#define _EXC_SVC_PRIO 0
-#define _IRQ_PRIO_OFFSET (_EXCEPTION_RESERVED_PRIO)
-#endif
+#define _EXC_SVC_PRIO (IS_ENABLED(CONFIG_ZERO_LATENCY_IRQS) ? 1 : 0)
+#define _IRQ_PRIO_OFFSET \
+	(_EXCEPTION_RESERVED_PRIO + IS_ENABLED(CONFIG_ZERO_LATENCY_IRQS))
```

Comment:
Would it be possible to use _EXC_SVC_PRIO in the last expression instead of
the IS_ENABLED(CONFIG_ZERO....)

Reply:
i think that IS_ENABLED(CONFIG_ZERO_LATENCY_IRQS)` suits better because it indicates that when ZLI is enabled then 1 additional level is reserved. In theory, _EXC_SVC_PRIO may be flipped with ZLI prio and set to 0, then offset would be wrong.

Reply:
> In theory, _EXC_SVC_PRIO may be flipped with ZLI prio and set to 0, then offset would be wrong.

Nope, this won't happen, right? This would defeat the purpose of the ZLI feature.

Reply:
I am suggesting :
`#define _IRQ_PRIO_OFFSET (_EXCEPTION_RESERVED_PRIO + _EXC_SVC_PRIO)`

This is more safe, i think, as we are free to set the _EXC_SVC_PRIO to whatever we want, and make sure the IRQ_PRIO_OFFSET is tracking this properly.

Reply:
I think this makes sense because the _exception_reserved_prio is reserved exactly for the SVC. Now, the SVC might need to be incremented by 1 due to ZLI, but the _IRQ_PRIO_OFFSET does not need to know about it

Reply:
I agree with @ioannisg here. `_IRQ_PRIO_OFFSET` is an integer value, so it is more readable to make it a sum of two values than a value and a result of a "boolean" expression. It also allows modifying `_EXC_SVC_PRIO` independently. Anyway, is the `(x) ? 1 : 0` above there to make code more readable?

Reply:
updated